classes/submissionFile/Repository.php

```diff
@@ -843,4 +849,59 @@ protected function getSubmissionFileLogData(SubmissionFile $submissionFile): arr
             'username' => $user?->getUsername(),
         ];
     }
+
+    /**
+     * Can be used to copy a SubmissionFile to another SubmissionFile along with the corresponding file
+     */
+    public function versionSubmissionFile(
+        SubmissionFile $submissionFile,
+        Publication $newPublication
+    ): SubmissionFile
+    {
+        $newSubmissionFile = clone $submissionFile;
+
+        $oldFileId = $submissionFile->getData('fileId');
+
+        $oldFile = Services::get('file')->get($oldFileId);
+
+        $submission = Repo::submission()->get($newPublication->getData('submissionId'));
+
+        $fileManager = new FileManager();
+        $extension = $fileManager->parseFileExtension($oldFile->path);
+
+        $submissionDir = Repo::submissionFile()
+            ->getSubmissionDir(
+                $submission->getData('contextId'),
+                $newPublication->getData('submissionId')
+            );
+
+        $newFileId = Services::get('file')->add(
+            Config::getVar('files', 'files_dir') . '/' . $oldFile->path,
+            $submissionDir . '/' . uniqid() . '.' . $extension
+        );
+
+        $newSubmissionFile->setData('id', null);
+        $newSubmissionFile->setData('assocId', $newPublication->getId());
+        $newSubmissionFile->setData('fileId', $newFileId);
+
+        $submissionFileId = Repo::submissionFile()
+            ->add($newSubmissionFile);
+
+        $submissionFile = Repo::submissionFile()
+            ->get($submissionFileId);
+
+        return $submissionFile;
+    }
+
+    /**
+     * returns jatsContent for Submission files that correspond to the content of the file
```

Comment:
I think this function can be removed. It shouldn't be necessary to use the `->summarize` to get the pathname (that data should be available directly from `$submissionFile`); the XML-specific parts of this (`jatsContent`, `jatsFileName`, `xmlString`) aren't actually specialized for JATS, and none of the other parts of the codebase (e.g. XML export) that need to read files from the disk use this. Plus I think you could get rid of the use of `$genres` entirely (also from calling code) if you didn't use `->summarize`.

Reply:
You are right that let's say `jatsContent` is not specific to the `JatsFile` except, in my opinion, when this content is auto generated by the submission metadata. When a file is uploaded, then the content is the submission file content. The same goes for the other `JatsFile` properties as well. 

After you pointed that out, that the `JatsFile` is not always a `SubmissionFile`, which makes perfect sense, I thought that keeping the `SubmissionFile` parent relationship on the `JatsFile` and the `SubmissionFile/Repository` parent relationship on the `Jats/Repository` could be generally misleading. To remedy that, I choose to drop the `JatsFile` relationships and keep the `JatsFile` the reference class for Jats actions/UI interactions. 

In that sense, in order to use this object to interact with `vue.js` I had to `summarize` it to the "appropriate" set of array properties, for the UI to work. I thought using a separate Schema for that, but I thought it does not fit here, because the `JatsFile` is not a "legit" entity but more of a conceptual placeholder for keeping the necessary exchange data together in one place.

Reply:
@asmecher you are right, I don't have to summarize the `$submissionFile` there in order to get the `path` that I need. 

Changed.